map/src/components/map.tsx

```diff
@@ -403,9 +430,35 @@ class Map extends React.Component<Props, {}> {
         }
 
         // Update labels of markers to be based on index in visibleMarkers
-        m.clustering.visibleMarkers.forEach((marker, index) => {
-          marker.setLabel((index + 1).toString());
-        });
+        m.clustering.visibleMarkers
+          .sort((a, b): number => {
+            const aPosition = a.getPosition();
+            const bPosition = b.getPosition();
+            const mapCenter = map.getCenter();
```

Comment:
Let's pull this out of the sort function to reduce the amount of calls to `getCenter()`